tests/formatters.test.js

```diff
@@ -0,0 +1,85 @@
+import * as formatters from '../src/lib/formatters';
+
+test('serializeFilters method should return the updated search params', () => {
+  const testUrl = new URL('http://test.html?timeStamp=null&id=["dummyId123"]');
+  const searchParams = new URLSearchParams(testUrl.search.slice(1));
+
+  const extIds = ['addon1@test.com', 'addon2@test.com'];
+  const viewTypes = ['popup', undefined];
+  const types = ['api_call', 'content_script'];
+  const names = ['test@api'];
+  const tabId = 22;
+  const keyword = 'test@substring';
+  const timeStamp = { start: 1597686226302, stop: 1597686226302 };
+
+  const updateFilter = {
+    id: { exclude: new Set(extIds) },
+    viewType: { exclude: new Set(viewTypes) },
+    type: { exclude: new Set(types) },
+    name: { exclude: new Set(names) },
+    tabId,
+    keyword,
+    timeStamp,
+  };
+
+  const serializedFormat = formatters.serializeFilters(
+    searchParams,
+    updateFilter
+  );
+
+  const expectedExtIds = '["addon1@test.com","addon2@test.com"]';
+  const expectedViewTypes = '["popup",null]';
+  const expectedTypes = '["api_call","content_script"]';
+  const expectedNames = '["test@api"]';
+  const expectedTabId = '22';
```

Comment:
This should have been `22`.

Use toStrictEqual instead of isEqual in your assertions to catch this kind of error.

Reply:
I am getting the `"22"` since I am getting the tab id from search params. Should I parse that into number and then check ?

Reply:
I don't think that you're getting `"22"` but 22. You're parsing `tabId` before returning it:

```javascript
const filterTabId = parseInt(searchParams.get('tabId'), 10) || null;
```

Reply:
> I don't think that you're getting `"22"` but 22. You're parsing `tabId` before returning it:
> 
> ```js
> const filterTabId = parseInt(searchParams.get('tabId'), 10) || null;
> ```

Yes, I have added this parsing now.